File: leetcode/2956-find-common-elements-between-two-arrays/findIntersectionValues.ts
function findIntersectionValues(nums1: number[], nums2: number[]): number[] {
  const result = [0, 0];

  for (const num1 of nums1) {
    for (const num2 of nums2) {
      if (num1 === num2) {
        result[0]++;
        break;
      }
    }
  }

  for (const num2 of nums2) {
    for (const num1 of nums1) {
      if (num1 === num2) {
        result[1]++;
        break;
      }
    }
  }

  return result;
}

export default findIntersectionValues;

File: leetcode/2956-find-common-elements-between-two-arrays/findIntersectionValues.go
package leetcode

func findIntersectionValues(nums1 []int, nums2 []int) []int {
	var result = make([]int, 2)

	for _, num1 := range nums1 {
		for _, num2 := range nums2 {
			if num1 == num2 {
				result[0]++
				break
			}
		}
	}

	for _, num2 := range nums2 {
		for _, num1 := range nums1 {
			if num1 == num2 {
				result[1]++
				break
			}
		}
	}

	return result
}
